### PR TITLE
Save description in HistoryEntry

### PR DESCRIPTION
**Pull request by Pierre-Monier**
## Related to [T115035](https://phabricator.wikimedia.org/T115035) and [T270229](https://phabricator.wikimedia.org/T270229)

I store the `PageTitle` description in local db, also add a test to check that every persisted properties of `PageTitle` are correctly set on the `HistoryEntry` instance.

Screenshot 2023-11-18 at 22 11 24
Screenshot 2023-11-18 at 22 10 47

### CODE REVIEW

**Comment by Pierre-Monier** [comment]
Working on a migration

**Comment by Pierre-Monier** [comment]
Migration is done, adding a default value to empty string for the new description column on the HistoryEntry object

File: app/src/main/java/org/wikipedia/history/HistoryEntry.kt [hunk 1]
```diff
@@ -30,10 +30,12 @@ class HistoryEntry(
     @Serializable(with = DateSerializer::class) var timestamp: Date = Date(),
     var source: Int = SOURCE_INTERNAL_LINK,
     var timeSpentSec: Int = 0,
+    var description: String = ""
 ) : Parcelable {
     constructor(title: PageTitle, source: Int, timestamp: Date = Date(), timeSpentSec: Int = 0) : this(title.wikiSite.authority(),
         title.wikiSite.languageCode, title.text, title.displayText, namespace = title.namespace,
-        timestamp = timestamp, source = source, timeSpentSec = timeSpentSec) {
+        timestamp = timestamp, source = source, timeSpentSec = timeSpentSec,
+        description = title.description ?: "") {
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/history/HistoryEntry.kt:38]: You can use `orEmpty()` here.

File: app/src/main/java/org/wikipedia/history/db/HistoryEntryWithImageDao.kt [hunk 2]
```diff
@@ -82,6 +82,10 @@ interface HistoryEntryWithImageDao {
             entryWithImage.displayTitle, 0, entryWithImage.namespace, entryWithImage.timestamp,
             entryWithImage.source, entryWithImage.timeSpentSec)
         entry.title.thumbUrl = entryWithImage.imageName
+
+        val hasDescription = entryWithImage.description?.isNotEmpty() ?: false
+        entry.title.description = if (hasDescription) entryWithImage.description else null
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/history/db/HistoryEntryWithImageDao.kt:87]: You can simply assign the value instead.

```
entry.title.description = entryWithImage.description
```

**Review by cooltey — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/history/db/HistoryEntryWithImageDao.kt (hunk 2 above)
Comment (Pierre-Monier) [reply app/src/main/java/org/wikipedia/history/db/HistoryEntryWithImageDao.kt:87]: Ok, I wasn't sure if it was a problem :)

**Comment by dbrant** [comment]
🎉 Thanks!

**Pull request merged**